src/metabase/api/embed.clj

```diff
@@ -274,16 +274,10 @@
   (run-query-for-unsigned-token (eu/unsign token) query-params))
 
 
-(api/defendpoint GET "/card/:token/query/csv"
-  "Like `GET /api/embed/card/query`, but returns the results as CSV."
-  [token & query-params]
-  (dataset-api/as-csv (run-query-for-unsigned-token (eu/unsign token) query-params, :constraints nil)))
-
-
-(api/defendpoint GET "/card/:token/query/json"
-  "Like `GET /api/embed/card/query`, but returns the results as JSOn."
-  [token & query-params]
-  (dataset-api/as-json (run-query-for-unsigned-token (eu/unsign token) query-params, :constraints nil)))
+(api/defendpoint GET "/card/:token/query/:export-format-name"
+  "Like `GET /api/embed/card/query`, but returns the results as a file in the specified format."
+  [token export-format-name & query-params]
```

Comment:
same thing, probably just shorten this to `export-format` and add a schema validation here